In [3]:
%load_ext autoreload
%autoreload 2

from src.base_classes.omic_data_loader import OmicDataLoader
from src.data_managers.concat import CatOmicDataManager

In [80]:
mrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/mrna",
)
mirna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/mirna_genes",
)
circrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/circrna",
)
# pirna_loader = OmicDataLoader(
#     data_dir="mds_data/splits_74/pirna",
# )
te_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/te_counts",
)

In [81]:
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}
odm = CatOmicDataManager(omic_data_loaders, n_splits=5)

save_folder = f"logs/mds_disease/mds_disease_{'_'.join(omic_data_loaders.keys())}.csv"
save_folder

'logs/mds_disease/mds_disease_mrna_mirna_circrna.csv'

In [ ]:
from src.evals.knn import KNNEvaluator

knn_eval = KNNEvaluator(
    data_manager=odm,
    n_trials=20,
    verbose=True,
    params={"k_lb": 1, "k_ub": 20},
)
_ = knn_eval.evaluate()
knn_eval.save_results(results_file=save_folder, row_name="knn")

[I 2024-11-14 13:01:00,120] A new study created in memory with name: no-name-e7bff091-bee4-44c3-94ba-29bdbe47c411
[I 2024-11-14 13:01:00,198] Trial 0 finished with value: 0.5085070524273005 and parameters: {'n_neighbors': 10}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,264] Trial 1 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 19}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,329] Trial 2 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 15}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,395] Trial 3 finished with value: 0.4477242436013849 and parameters: {'n_neighbors': 11}. Best is trial 0 with value: 0.5085070524273005.


New best score: 0.509
Best model performance:
Accuracy: 0.879 ± 0.025
F1 Macro: 0.682 ± 0.116
F1 Weighted: 0.848 ± 0.038


[I 2024-11-14 13:01:00,464] Trial 4 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 14}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,527] Trial 5 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 19}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,590] Trial 6 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 19}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,652] Trial 7 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 17}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,714] Trial 8 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 14}. Best is trial 0 with value: 0.5085070524273005.
[I 2024-11-14 13:01:00,776] Trial 9 finished with value: 0.6139268814359082 and parameters: {'n_neighbors': 3}. Best is trial 9 with value: 0.6139268814359082.
[I 2024-11-14 13:01:00,875] Tr

New best score: 0.614
Best model performance:
Accuracy: 0.892 ± 0.032
F1 Macro: 0.782 ± 0.064
F1 Weighted: 0.880 ± 0.041


[I 2024-11-14 13:01:01,007] Trial 12 finished with value: 0.46772551930590295 and parameters: {'n_neighbors': 7}. Best is trial 9 with value: 0.6139268814359082.
[I 2024-11-14 13:01:01,071] Trial 13 finished with value: 0.5296409475158396 and parameters: {'n_neighbors': 8}. Best is trial 9 with value: 0.6139268814359082.
[I 2024-11-14 13:01:01,135] Trial 14 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 6}. Best is trial 14 with value: 0.6528081564994511.
[I 2024-11-14 13:01:01,199] Trial 15 finished with value: 0.6139268814359082 and parameters: {'n_neighbors': 3}. Best is trial 14 with value: 0.6528081564994511.
[I 2024-11-14 13:01:01,264] Trial 16 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 5}. Best is trial 14 with value: 0.6528081564994511.
[I 2024-11-14 13:01:01,334] Trial 17 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 6}. Best is trial 14 with value: 0.6528081564994511.


New best score: 0.653
Best model performance:
Accuracy: 0.907 ± 0.053
F1 Macro: 0.806 ± 0.106
F1 Weighted: 0.893 ± 0.062


[I 2024-11-14 13:01:01,403] Trial 18 finished with value: 0.5085070524273005 and parameters: {'n_neighbors': 10}. Best is trial 14 with value: 0.6528081564994511.
[I 2024-11-14 13:01:01,469] Trial 19 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 5}. Best is trial 14 with value: 0.6528081564994511.
[I 2024-11-14 13:01:01,470] A new study created in memory with name: no-name-39c8076c-fcec-4692-ab9d-dba5ce98c8d0
[I 2024-11-14 13:01:01,541] Trial 0 finished with value: 0.5809969155423703 and parameters: {'C': 0.027203803863838624, 'class_weight': 'balanced', 'rfe_step': 0.16047342946164844, 'rfe_n_features': 181}. Best is trial 0 with value: 0.5809969155423703.
[I 2024-11-14 13:01:01,615] Trial 1 finished with value: 0.5614209168802691 and parameters: {'C': 0.01808785237908672, 'class_weight': 'balanced', 'rfe_step': 0.10392392789249999, 'rfe_n_features': 176}. Best is trial 0 with value: 0.5809969155423703.
[I 2024-11-14 13:01:01,675] Trial 2 finished with value:

New best score: 0.581
Best model performance:
Accuracy: 0.867 ± 0.084
F1 Macro: 0.775 ± 0.120
F1 Weighted: 0.865 ± 0.079


[I 2024-11-14 13:01:01,794] Trial 3 finished with value: 0.7389770960193865 and parameters: {'C': 0.25817777503003897, 'class_weight': None, 'rfe_step': 0.13820158797311796, 'rfe_n_features': 119}. Best is trial 3 with value: 0.7389770960193865.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterati

New best score: 0.739
Best model performance:
Accuracy: 0.933 ± 0.042
F1 Macro: 0.857 ± 0.094
F1 Weighted: 0.924 ± 0.051


[I 2024-11-14 13:01:02,145] Trial 5 finished with value: 0.7039337563024542 and parameters: {'C': 0.6319302254447319, 'class_weight': 'balanced', 'rfe_step': 0.11227973720404164, 'rfe_n_features': 126}. Best is trial 3 with value: 0.7389770960193865.
[I 2024-11-14 13:01:02,274] Trial 6 finished with value: 0.7389770960193865 and parameters: {'C': 0.8660283351283377, 'class_weight': None, 'rfe_step': 0.16250197355361073, 'rfe_n_features': 164}. Best is trial 3 with value: 0.7389770960193865.
[I 2024-11-14 13:01:02,497] Trial 7 finished with value: 0.669993140387569 and parameters: {'C': 0.19390436048362267, 'class_weight': 'balanced', 'rfe_step': 0.06296156664023142, 'rfe_n_features': 111}. Best is trial 3 with value: 0.7389770960193865.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-pac

New best score: 0.753
Best model performance:
Accuracy: 0.933 ± 0.042
F1 Macro: 0.870 ± 0.092
F1 Weighted: 0.927 ± 0.052


[I 2024-11-14 13:01:06,171] Trial 3 finished with value: 0.7389770960193865 and parameters: {'lambda': 0.01694761860106469, 'alpha': 1.4864468309677284e-05}. Best is trial 0 with value: 0.7526861283892176.
[I 2024-11-14 13:01:06,259] Trial 4 finished with value: 0.553150014770979 and parameters: {'lambda': 0.030960126688156214, 'alpha': 0.0473458448805854}. Best is trial 0 with value: 0.7526861283892176.
[I 2024-11-14 13:01:06,348] Trial 5 finished with value: 0.27794757275052406 and parameters: {'lambda': 2.6701102970387854e-07, 'alpha': 0.6694793346250602}. Best is trial 0 with value: 0.7526861283892176.
[I 2024-11-14 13:01:06,454] Trial 6 finished with value: 0.7389770960193865 and parameters: {'lambda': 2.204321254027931e-06, 'alpha': 0.0264702001022464}. Best is trial 0 with value: 0.7526861283892176.
[I 2024-11-14 13:01:06,553] Trial 7 finished with value: 0.553150014770979 and parameters: {'lambda': 0.04626876857643086, 'alpha': 0.042348611253685685}. Best is trial 0 with value:

New best score: 0.797
Best model performance:
Accuracy: 0.947 ± 0.050
F1 Macro: 0.894 ± 0.106
F1 Weighted: 0.941 ± 0.059


[I 2024-11-14 13:01:09,873] Trial 40 finished with value: 0.7526861283892176 and parameters: {'lambda': 6.052428990708667e-05, 'alpha': 1.1933602260962324e-05}. Best is trial 37 with value: 0.7966829243918474.
[I 2024-11-14 13:01:09,978] Trial 41 finished with value: 0.7526861283892176 and parameters: {'lambda': 4.3825441752949765e-06, 'alpha': 2.1328527830188963e-08}. Best is trial 37 with value: 0.7966829243918474.
[I 2024-11-14 13:01:10,074] Trial 42 finished with value: 0.7526861283892176 and parameters: {'lambda': 2.872659858507051e-05, 'alpha': 4.377963266123595e-07}. Best is trial 37 with value: 0.7966829243918474.
[I 2024-11-14 13:01:10,169] Trial 43 finished with value: 0.7526861283892176 and parameters: {'lambda': 0.00029039857179555636, 'alpha': 6.607391893569826e-08}. Best is trial 37 with value: 0.7966829243918474.
[I 2024-11-14 13:01:10,260] Trial 44 finished with value: 0.7526861283892176 and parameters: {'lambda': 1.8588986711453626e-06, 'alpha': 2.15154443593222e-07}. 

New best score: 0.704
Best model performance:
Accuracy: 0.920 ± 0.050
F1 Macro: 0.840 ± 0.097
F1 Weighted: 0.911 ± 0.056


`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-pa

Results saved to logs/mds_disease/mds_disease_te.csv

Epoch: 101:
Train Loss: 0.2859, Train Acc: 0.9322, Train F1 Macro: 0.8554, Train F1 Weighted: 0.9251
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026
Test Acc: 0.8667, Test F1 Macro: 0.7115, Test F1 Weighted: 0.8385
##################################################

Epoch: 101:
Train Loss: 0.2600, Train Acc: 0.9322, Train F1 Macro: 0.8554, Train F1 Weighted: 0.9251
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026
Test Acc: 0.8667, Test F1 Macro: 0.7115, Test F1 Weighted: 0.8385
##################################################

Epoch: 101:
Train Loss: 0.3707, Train Acc: 0.8475, Train F1 Macro: 0.5489, Train F1 Weighted: 0.7915
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026
Test Acc: 0.8667, Test F1 Macro: 0.7115, Test F1 Weighted: 0.8385
##################################################

Epoch: 101:
Trai

[I 2024-11-14 13:01:32,206] Trial 0 finished with value: 0.6862235118921005 and parameters: {}. Best is trial 0 with value: 0.6862235118921005.
[I 2024-11-14 13:01:32,246] A new study created in memory with name: no-name-dc528e16-61f4-4f53-b290-a7c4cf544e9d


New best score: 0.686
Best model performance:
Accuracy: 0.920 ± 0.050
F1 Macro: 0.824 ± 0.109
F1 Weighted: 0.906 ± 0.061

Epoch: 050:
Train Loss: 0.4637, Train Acc: 0.8305, Train F1 Macro: 0.4537, Train F1 Weighted: 0.7536
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111
##################################################

Epoch: 100:
Train Loss: 0.4587, Train Acc: 0.8305, Train F1 Macro: 0.4537, Train F1 Weighted: 0.7536
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111
##################################################

Epoch: 150:
Train Loss: 0.4380, Train Acc: 0.8305, Train F1 Macro: 0.4537, Train F1 Weighted: 0.7536
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111


[I 2024-11-14 13:01:48,214] Trial 0 finished with value: 0.3717283965781297 and parameters: {}. Best is trial 0 with value: 0.3717283965781297.



Epoch: 200:
Train Loss: 0.5122, Train Acc: 0.8167, Train F1 Macro: 0.4495, Train F1 Weighted: 0.7343
Val Acc: 0.8571, Val F1 Macro: 0.4615, Val F1 Weighted: 0.7912, Val Geometric Mean: 0.6790
Test Acc: 0.8571, Test F1 Macro: 0.4615, Test F1 Weighted: 0.7912
##################################################
New best score: 0.372
Best model performance:
Accuracy: 0.811 ± 0.077
F1 Macro: 0.586 ± 0.116
F1 Weighted: 0.781 ± 0.067


In [ ]:
from src.evals.svm import SVMEvaluator

svm_eval = SVMEvaluator(
    data_manager=odm,
    n_trials=30,
    verbose=True,
    params={
        "C_lb": 0.01,
        "C_ub": 10,
        "rfe_step_range": (0.05, 0.2),
        "rfe_n_features_range": (100, 200),
    },
    mode="linear",
)
_ = svm_eval.evaluate()
svm_eval.save_results(results_file=save_folder, row_name="svm")

In [82]:
from src.evals.xgboost import XGBoostEvaluator

xgb_eval = XGBoostEvaluator(
    data_manager=odm,
    n_trials=60,
    verbose=True,
)
_ = xgb_eval.evaluate()
xgb_eval.print_best_results()
# xgb_eval.save_results(results_file=save_folder, row_name="xgb")

[I 2024-11-17 17:49:43,818] A new study created in memory with name: no-name-15fc25c5-235a-413c-9c09-c3b99f2534ff
[I 2024-11-17 17:49:43,979] Trial 0 finished with value: 0.806425114231087 and parameters: {'lambda': 0.0005522152199585571, 'alpha': 1.5631575756388323e-06}. Best is trial 0 with value: 0.806425114231087.
[I 2024-11-17 17:49:44,126] Trial 1 finished with value: 0.806425114231087 and parameters: {'lambda': 0.0009441682055922977, 'alpha': 1.1298850310301638e-07}. Best is trial 0 with value: 0.806425114231087.


New best score: 0.806
Best model performance:
Accuracy: 0.946 ± 0.027
F1 Macro: 0.904 ± 0.048
F1 Weighted: 0.944 ± 0.028


[I 2024-11-17 17:49:44,285] Trial 2 finished with value: 0.43192671408544425 and parameters: {'lambda': 0.004917887497782284, 'alpha': 0.11819472221563164}. Best is trial 0 with value: 0.806425114231087.
[I 2024-11-17 17:49:44,500] Trial 3 finished with value: 0.6460639796087643 and parameters: {'lambda': 3.5013820406033616e-06, 'alpha': 0.018691467676913313}. Best is trial 0 with value: 0.806425114231087.
[I 2024-11-17 17:49:44,654] Trial 4 finished with value: 0.7948768203234714 and parameters: {'lambda': 0.008328351583202227, 'alpha': 2.072834250567776e-06}. Best is trial 0 with value: 0.806425114231087.
[I 2024-11-17 17:49:44,800] Trial 5 finished with value: 0.7039337563024542 and parameters: {'lambda': 0.23693265707256447, 'alpha': 1.2992226851648535e-08}. Best is trial 0 with value: 0.806425114231087.
[I 2024-11-17 17:49:44,945] Trial 6 finished with value: 0.6969195446998454 and parameters: {'lambda': 0.0264117036771058, 'alpha': 0.012315385529456572}. Best is trial 0 with valu

New best score: 0.813
Best model performance:
Accuracy: 0.946 ± 0.027
F1 Macro: 0.909 ± 0.047
F1 Weighted: 0.945 ± 0.028


[I 2024-11-17 17:49:46,389] Trial 15 finished with value: 0.7693788056664252 and parameters: {'lambda': 0.00016310395977176353, 'alpha': 6.635973437137921e-06}. Best is trial 13 with value: 0.8127007161803061.
[I 2024-11-17 17:49:46,586] Trial 16 finished with value: 0.7039337563024542 and parameters: {'lambda': 0.9772311749606422, 'alpha': 0.00036215662812966583}. Best is trial 13 with value: 0.8127007161803061.
[I 2024-11-17 17:49:46,748] Trial 17 finished with value: 0.6684310669837741 and parameters: {'lambda': 4.133461888317546e-07, 'alpha': 6.681742465534682e-08}. Best is trial 13 with value: 0.8127007161803061.
[I 2024-11-17 17:49:46,906] Trial 18 finished with value: 0.7699660701774514 and parameters: {'lambda': 0.00011760138805022151, 'alpha': 2.440916720271865e-06}. Best is trial 13 with value: 0.8127007161803061.
[I 2024-11-17 17:49:47,066] Trial 19 finished with value: 0.6969195446998454 and parameters: {'lambda': 0.0003333712475177214, 'alpha': 4.2055012276993824e-05}. Bes

New best score: 0.852
Best model performance:
Accuracy: 0.960 ± 0.033
F1 Macro: 0.928 ± 0.059
F1 Weighted: 0.957 ± 0.035


[I 2024-11-17 17:49:49,534] Trial 34 finished with value: 0.806425114231087 and parameters: {'lambda': 0.0041929710964316805, 'alpha': 8.728473616443046e-07}. Best is trial 32 with value: 0.8523939840000002.
[I 2024-11-17 17:49:49,717] Trial 35 finished with value: 0.8523255135749392 and parameters: {'lambda': 0.0003730328218059249, 'alpha': 3.4751259612204713e-06}. Best is trial 32 with value: 0.8523939840000002.
[I 2024-11-17 17:49:49,877] Trial 36 finished with value: 0.806425114231087 and parameters: {'lambda': 0.00033894234388439265, 'alpha': 4.946794044412732e-06}. Best is trial 32 with value: 0.8523939840000002.
[I 2024-11-17 17:49:50,040] Trial 37 finished with value: 0.7013240844574544 and parameters: {'lambda': 1.7274186184011694e-05, 'alpha': 3.9886668717947476e-07}. Best is trial 32 with value: 0.8523939840000002.
[I 2024-11-17 17:49:50,199] Trial 38 finished with value: 0.7699660701774514 and parameters: {'lambda': 9.467265638899228e-05, 'alpha': 2.0189527871812583e-06}. B

Best model performance:
Accuracy: 0.960 ± 0.033
F1 Macro: 0.928 ± 0.059
F1 Weighted: 0.957 ± 0.035


In [ ]:
from src.evals.mlp import MLPEvaluator

mlp_eval = MLPEvaluator(
    data_manager=odm,
    n_trials=10,
    verbose=True,
    params={
        "lr_range": [1e-4, 1e-2],
        "l2_lambda": 5e-4,
        "dropout_range": [0.1, 0.6],
        "hidden_channels": 64,
        "proj_dim": 64,
        "batch_size": 32,
        "max_epochs": 50,
    },
)
_ = mlp_eval.evaluate()
# mlp_eval.save_results(results_file=save_folder, row_name="mlp")

[I 2024-11-17 12:46:25,884] A new study created in memory with name: no-name-a09d1753-b355-4d8b-9dab-5aa06e28e0d1
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=100` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using

New best score: 0.663
Best model performance:
Accuracy: 0.906 ± 0.032
F1 Macro: 0.815 ± 0.062
F1 Weighted: 0.898 ± 0.038


`Trainer.fit` stopped: `max_epochs=100` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=100` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-

In [99]:
mlp_eval.print_best_results()
mlp_eval.print_best_parameters()

Best model performance:
Accuracy: 0.906 ± 0.032
F1 Macro: 0.815 ± 0.062
F1 Weighted: 0.898 ± 0.038
Best hyperparameters:
{'lr': 0.004521333766373588, 'dropout': 0.5753805682085522}


In [98]:
from src.evals.mogonet import MOGONETEvaluator
from src.data_managers.sample_graph import SampleGraphDataManager

mogonet_eval = MOGONETEvaluator(
    data_manager=SampleGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "graph_style": "threshold",
            "self_connections": True,
            "avg_degree": 5,
        },
    ),
    n_trials=1,
    params={
        "encoder_hidden_channels": {
            "mrna": 64,
            "mirna": 64,
            "circrna": 64,
            # "te": 64,
        },
        "encoder_type": "gat",
        "dropout": 0.2,
        "integrator_type": "attention", # vcdn seems to work better on mds disease
        "integration_in_dim": 8,
        "vcdn_hidden_channels": 8,
        "epochs": 250,
        "log_interval": 251,
    }
)
mogonet_eval.evaluate()
mogonet_eval.print_best_results()
mogonet_eval.save_results(results_file=save_folder, row_name="mogonet")

[I 2024-11-17 17:59:30,901] A new study created in memory with name: no-name-06dadf4f-1116-41e1-b3a1-f279da0538d6


Using: attention integrator
Using: attention integrator
Using: attention integrator
Using: attention integrator
Using: attention integrator


[I 2024-11-17 17:59:44,598] Trial 0 finished with value: 0.8999268693333332 and parameters: {}. Best is trial 0 with value: 0.8999268693333332.


New best score: 0.900
Best model performance:
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
Best model performance:
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035


# linear
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
# vcdn
- integration dim = 2
Accuracy: 0.825 ± 0.030
F1 Macro: 0.452 ± 0.009
F1 Weighted: 0.746 ± 0.043
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.973 ± 0.033
F1 Macro: 0.957 ± 0.053
F1 Weighted: 0.973 ± 0.033
- integration dim = 16
Accuracy: 0.973 ± 0.053
F1 Macro: 0.958 ± 0.083
F1 Weighted: 0.973 ± 0.053
# attention - faster than vcdn
- integration dim = 2
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
- integration dim = 16
Accuracy: 0.960 ± 0.053
F1 Macro: 0.934 ± 0.085
F1 Weighted: 0.959 ± 0.054

In [79]:
from src.evals.birgat import BiRGATEvaluator
from src.data_managers.bipartite_graph import BipartiteGraphDataManager

omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                # "te": 1.8,
            },
        },
    ),
    params={
        "epochs": 250,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": False,
    },
    n_trials=1,
)
birgat_eval.evaluate()
birgat_eval.print_best_results()
# birgat_eval.save_results(results_file=save_folder, row_name="birgat")

isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)


[I 2024-11-17 17:47:24,249] A new study created in memory with name: no-name-8982798b-dd55-4149-a42e-5888ce9dfb5d


isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)

Epoch: 050:
Train Loss: 0.2900, Train Acc: 0.8136, Train F1 Macro: 0.4486, Train F1 Weighted: 0.7451
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111
##################################################

Epoch: 100:
Train Loss: 0.1890, Train Acc: 0.9322, Train F1 Macro: 0.8883, Train F1 Weighted: 0.9346
Val Acc: 0.8667, Val F1 Macro: 0.7115, Val F1 Weighted: 0.8385, Val Geometric Mean: 0.8026
Test Acc: 0.8667, Test F1 Macro: 0.7115, Test F1 Weighted: 0.8385
##################################################

Epoch: 150:
Train Loss: 0.1008, Train Acc: 0.9322, Train F1 Macro: 0.8554, Train F1 Weighted: 0.9

[I 2024-11-17 17:48:28,504] Trial 0 finished with value: 0.8666250016074483 and parameters: {}. Best is trial 0 with value: 0.8666250016074483.



Epoch: 250:
Train Loss: 0.0000, Train Acc: 1.0000, Train F1 Macro: 1.0000, Train F1 Weighted: 1.0000
Val Acc: 0.7143, Val F1 Macro: 0.4167, Val F1 Weighted: 0.7143, Val Geometric Mean: 0.5968
Test Acc: 0.7143, Test F1 Macro: 0.4167, Test F1 Weighted: 0.7143
##################################################
New best score: 0.867
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053


```python
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                # "te": 1.8,
            },
        },
    ),
    params={
        "epochs": 250,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": False,
    },
    n_trials=1,
)
```

Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053

# mrna, mirna, circrna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, circrna, 2L no interactions
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064
---
Accuracy: 0.946 ± 0.027
F1 Macro: 0.904 ± 0.048
F1 Weighted: 0.944 ± 0.028
# mrna, mirna, circrna 3L, interactions, degree ~20 in diff exp graphs, larger degree shows degraded performance
# making the avg degree to high shows large jumps on the validation set during training
Accuracy: 0.945 ± 0.053
F1 Macro: 0.910 ± 0.081
F1 Weighted: 0.946 ± 0.048
# mrna, mirna, circrna 2L, interactions, 64 cap
Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053
# mrna, mirna, circrna 3L, interactions, 64 cap
Accuracy: 0.891 ± 0.054
F1 Macro: 0.801 ± 0.088
F1 Weighted: 0.888 ± 0.056
# mrna, mirna, circrna 3L
Accuracy: 0.920 ± 0.050
F1 Macro: 0.829 ± 0.112
F1 Weighted: 0.907 ± 0.062
# mrna, mirna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, 3L
Accuracy: 0.947 ± 0.050
F1 Macro: 0.897 ± 0.089
F1 Weighted: 0.944 ± 0.051